pkg/tsdb/loki/loki.go

```diff
@@ -167,7 +169,8 @@ func queryData(ctx context.Context, req *backend.QueryDataRequest, dsInfo *datas
 		return result, err
 	}
 
-	for _, query := range queries {
+	err = concurrency.ForEachJob(ctx, len(queries), 10, func(ctx context.Context, idx int) error {
```

Comment:
This is a useful library I didn't know about :-)
I think the concurrency limit (`10`) should be configurable. The reason is that be changing this we are removing a "partial rate limiter" (partial because this is all within a query). But the net impact may be more concurrent queries to the underlying data store (loki in this case). This has the potential to have negative consequences in certain installations, so we should allow administrators to control it.

Reply:
I imagine this is less likely a problem with Loki in particular, but I imagine people will copy this pattern into datasources.

Reply:
> I think the concurrency limit (10) should be configurable. 

This makes sense. 10 was a best guess on what's reasonable to issue to Loki in Parallel. 


The other big issue in porting more widely is shared state. We can't _necessarily_ just put a `ForEachJob` on it and be done. In this case `api` is a pointer shared across the jobs, but it didn't look like any state changes in it so it ought to be good.